### PR TITLE
Node.js - Frontend or Backend?

### PR DESCRIPTION
**Pull request by hectorkambow**
Duplicate of this [PR ](https://github.com/section-io/engineering-education/pull/254/files?short_path=36ff85d#diff-36ff85d7e5cec03229130389a6ddd399)

### CODE REVIEW

**Comment by hectorkambow** [comment]
@lalith1403 this is the duplicate PR I created

File: articles/nodejs-frontend-backend/index.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    alt: front end and backend with Node.js
+---
+Let us start by understanding what Node.js is. Node.js is a runtime environment for executing server-side code with higher efficiency and it presents a larger bandwidth to handle large code payloads. The objective of this article is as follows-
+<!--more-->
+
+1. Understand and clarify any doubt of whether Node.js is frontend or backend application
+2. List out various applications in both frontend and backend
+3. Understand how Node.js complements other technologies in the frontend and backend
+
+The choice of the language that Ryan Dahl, the creator of Node.js, opted for is the most popular programming language, JavaScript.
```
Comment (hectorkambow) [inline articles/nodejs-frontend-backend/index.md:24]: would be helpful to add a source - anything that is not common language lets add a source @lalith1403

File: articles/nodejs-frontend-backend/index.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+Node.js increased the overall control developers had over applications. Developers exercise a greater level of control over system calls, database calls, and network calls. Hence, we achieve better performing and feature-rich web applications. JavaScript is still the most preferred programming language across the world. StackOverflow survey results prove the same. This feat achieved by JavaScript enables faster progress for all the libraries and frameworks associated with JS, including node.js.
```
Comment (hectorkambow) [inline articles/nodejs-frontend-backend/index.md:25]: same the claim JS is the most popular - could use a source to help back up your claim

File: articles/nodejs-frontend-backend/index.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+![](/engineering-education/nodejs-frontend-backend/stackoverflowresults.jpg)
+[Image Source](https://insights.stackoverflow.com/survey/2020)
+
+<!-- Frontend -->
+### The Question: Frontend or Backend?
+
+How do we analyze the question? Node.js set out as a server-side engine. There are a massive number of libraries built for Node.js.
```
Comment (hectorkambow) [inline articles/nodejs-frontend-backend/index.md:33]: May be just me - but I don't quite follow the second sentence - Node.js set out as a server-side engine? 
Did you mean to say Node.js set out to be a server side engine?

**Comment by hectorkambow** [comment]
👋  Good afternoon Shaun - wanted to ask if you could give this Node.js (Backend or Frontend) article a tech review before we publish - when you have a moment. 

Thank you 🚀

File: articles/nodejs-frontend-backend/index.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+![](/engineering-education/nodejs-frontend-backend/stackoverflowresults.jpg)
+[Image Source](https://insights.stackoverflow.com/survey/2020)
+
+<!-- Frontend -->
+### The Question: Frontend or Backend?
+How do we analyze the question? Node.js set out as a server-side engine. There is a massive number of libraries built for Node.js.
+The idea of using Node.js for frontend is a natural extension of the various features that it provides. Let us understand why the use of Node.js enhances the developer experience.
+
+1. JavaScript: The ability to use the same language for both frontend and backend makes it easier for the developer.
+2. Availability of Packages and Libraries: There are many libraries available for use in the frontend. For example, the 'moment' is a library that processes date in the specified format.
```
Comment (interludic2000) [inline articles/nodejs-frontend-backend/index.md:36]: Link to moment.js site

File: articles/nodejs-frontend-backend/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+Now we have a better understanding whether frontend applications can be used with Node.js, the answer is a huge YES. Yes, Node.js is used in both the frontend and backend applications. Let us now dive into the applications that Node.js supports in the frontend and backend.
+
+### Frontend
+1. **Code Processors**: Code processors are required for displaying text in the browser. The HTML and CSS files that we receive are usually minimized in size by removing all the unnecessary characters for faster rendering and transmission. The pre-processors help the browser in rendering the files. Node.js enables us to code these processors in JS, thus allowing larger flexibility and customizability.
+
+2. **Code Linters**: A linter is a program that helps us identify and correct issues in our codebase. Issues related to syntax, certain custom standards specified by the development team, and programming errors can be identified using linters. Custom linters enhance the overall efficiency of the dev team at a company. One such example is the ESLint written in Node.js. Linters help out with interpreted languages, such as JavaScript, since the compilation step does not exist.
```
Comment (interludic2000) [inline articles/nodejs-frontend-backend/index.md:44]: Link to ESLint

File: articles/nodejs-frontend-backend/index.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+With the help of a linter, errors are caught while coding. The list of errors can be found at the [official documentation](https://eslint.org/docs/rules/) for ESLint. Custom linters can be built on top of the ESLint. The Airbnb configuration, developed by Airbnb for internal software development, is open source. It is used by the majority of JS developers across the globe.
```
Comment (interludic2000) [inline articles/nodejs-frontend-backend/index.md:46]: Indentation looks wrong on this paragraph

File: articles/nodejs-frontend-backend/index.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+3. **Module Bundlers**: Module bundlers are programs that take in various code files and bundles them into a single file. Such programs are usually included with Web frameworks like React. [Webpack](https://en.wikipedia.org/wiki/Webpack) is another module bundler, it is a static module bundler built on top of Node.js.
+
+4. **Styling**: Styling the webpages is usually done with CSS. Packages such as styled-components have made styling very easy. Styled-components is a library written for React.js. It allows easier integration of styling with JS, resulting in more efficient styling code for user interaction.
```
Comment (interludic2000) [inline articles/nodejs-frontend-backend/index.md:50]: link to Styled-components

File: articles/nodejs-frontend-backend/index.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+1. **Network and API calls**: We looked at the various libraries available that allow us to make API calls, in this [previous article](/engineering-education/http-requests-nodejs/). Have a look at it to get a better understanding into HTTP requests. Node.js gives us a greater degree of control over the HTTP requests made. We can specify the headers, set up proxies, and receive responses in formats suitable to our application.   
+2. **Database Integration**: Node.js has libraries and interfaces that allow seamless integration with databases. CRUD (Create, Read, Update, and Delete) operations are done using JS. Database integrations allow us to handle database operations in JavaScript, making the learning curve for databases a lot easier to grasp.   
+3. **Operating System-Level Control over Application**: System level programming, considers the computational resources a program will use and it can be controlled using node.js. JavaScript provides features for concurrency and OS programming.
```
Comment (interludic2000) [inline articles/nodejs-frontend-backend/index.md:61]: > controlled using node.js

Change node.js to uppercase Node.js

File: articles/nodejs-frontend-backend/index.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+4. **Real-Time Applications**: Let us look at some examples of real-time applications. They range from software solutions like live streaming, real-time logistics, tracking, social networking platforms, and hardware solutions such as Internet Of Things(IoT) applications. All the features discussed above help in designing systems that are scalable, feature-rich. The advantages of having the codebase predominantly in one language enable developers to debug errors easily and build systems that are easily integrable with each other.
```
Comment (interludic2000) [inline articles/nodejs-frontend-backend/index.md:62]: > are scalable, feature-rich.

are scalable and feature-rich.

**Pull request merged**